x-pack/plugins/monitoring/public/services/enable_alerts_modal.js

```diff
@@ -12,12 +12,10 @@ export function enableAlertsModalProvider($http, $window, $injector) {
     const modalHasBeenShown = $window.sessionStorage.getItem('ALERTS_MODAL_HAS_BEEN_SHOWN');
     const decisionMade = $window.localStorage.getItem('ALERTS_MODAL_DECISION_MADE');
 
-    if (Object.keys(alerts.allAlerts).length > 0) {
+    if (Object.keys(alerts) > 0) {
       $window.localStorage.setItem('ALERTS_MODAL_DECISION_MADE', true);
```

Comment:
You have defined later on a function `notAskAgain` that updates the localStorage. How about calling that function here?

Reply:
Actually wouldn't it make more sense to completely delete the line that sets the localStorage and move it to the place where `shouldShowAlertsModal` is being called? shouldShowAlertsModal should return a boolean and do nothing more. `enable_alerts_modal.tsx` that calls this function should then call the `notAskAgain` function to update the localStorage. Thus being said the code could be refactored to something like this:

```suggestion
      function shouldShowAlertsModal(alerts) {
    const modalHasBeenShown = $window.sessionStorage.getItem('ALERTS_MODAL_HAS_BEEN_SHOWN');
    const decisionMade = $window.localStorage.getItem('ALERTS_MODAL_DECISION_MADE');

    if (Object.keys(alerts).length === 0 && (!modalHasBeenShown || !decisionMade)) {
      return true;
    }

    return false;
  }
```

* I put an `||` instead of `&&` here (!modalHasBeenShown || !decisionMade) because it makes more sense with the test case you defined `When the user doesn't have existing rules and selects "Remind me later" or closes the modal with the X button:`